nifi-nar-bundles/nifi-kudu-bundle/nifi-kudu-processors/src/main/java/org/apache/nifi/processors/kudu/PutKudu.java

```diff
@@ -348,6 +397,14 @@ private void trigger(final ProcessContext context, final ProcessSession session,
         session.adjustCounter("Records Inserted", totalCount, false);
     }
 
+    private String getEvaluatedProperty(PropertyDescriptor property, ProcessContext context, FlowFile flowFile) {
```

Comment:
I believe that with this approach a flow file not having the expected attribute will be sent back to the input relationship and the processor will keep processing the same flow file and failing. It should probably be handled separately and sent to the failure relationship because it'll always fail with this "bad" flow file.

Reply:
This method is called in a try/catch that marks the flowfile as failed and then transfers it to the failure relationship. 

```
...
} catch (Exception ex) {
   flowFileFailures.put(flowFile, ex);
}             
...
if (flowFileFailures.containsKey(flowFile)) {
   getLogger().error("Failed to write due to {}", new Object[]{flowFileFailures.get(flowFile)});
   session.transfer(flowFile, REL_FAILURE);
...
```

Does that result in the correct behavior?

Reply:
Hmm this method is called on L270-274 with is outside the ``try {} catch (){}`` starting L276, no?

Reply:
oh, yeah not sure how I missed that. I will move them into the try/catch.